django/db/models/fields/generated.py

```diff
@@ -33,6 +37,11 @@ def __init__(self, *, expression, db_persist=None, output_field=None, **kwargs):
         self.db_persist = db_persist
         super().__init__(**kwargs)
 
+    def _check_output_field(self, **kwargs):
+        if isinstance(self.output_field, CharField):
+            return self.output_field.check(**kwargs)
+        return []
+
```

Comment:
It should be called from `self.check` and not be dependent of the type of `self.output_field`.

Reply:
So all I have to do in `_check_output_field` is this?
```
def _check_output_field(self, **kwargs):
        return self.output_field.check()
```

Reply:
Do I need to call `self.output_field.check()` only? 
I am asking because I think it checks if the name for the field is valid as well, which in this case it doesn't have?


In `Fields.check`
```py
def check(self, **kwargs):
        return [
            *self._check_field_name(),  # <-- this line
            *self._check_choices(),
            *self._check_db_default(**kwargs),
...
```

Reply:
Hello, Simon!
I did this and now receiving the message to specify `max_length`

```py
def check(self, **kwargs):
      databases = kwargs.get("databases") or []
      return [
          *super().check(**kwargs),
          *self._check_output_field(**kwargs),
          *self._check_supported(databases),
          *self._check_persistence(databases),
      ]

def _check_output_field(self, **kwargs):
        # Ignore field name checks as the output_field is not a field.
        self.output_field.name = self.name + "_output_field"
        self.output_field.model = self.model
        return self.output_field.check(**kwargs)
```

Reply:
@charettes haven't heard back from you? 🙂